src/nvim/syntax.c

```diff
@@ -7262,6 +7265,28 @@ int syn_get_final_id(int hl_id)
   return hl_id;
 }
 
+/* Refresh the color attributes of all highlight
+ * groups.
+ * This usually needs to be done after the "Normal"
+ * group is modified to update the groups that might
+ * be using "bg" or "fg".
+ */
+void highlight_groups_refresh(void)
+{
+  int idx;
+  for (idx = 0; idx < highlight_ga.ga_len; idx++) {
+    if (HL_TABLE()[idx].sg_rgb_bg_name != NULL) {
+      HL_TABLE()[idx].sg_rgb_bg = name_to_color(HL_TABLE()[idx].sg_rgb_bg_name);
+    }
+    if (HL_TABLE()[idx].sg_rgb_fg_name != NULL) {
+      HL_TABLE()[idx].sg_rgb_fg = name_to_color(HL_TABLE()[idx].sg_rgb_fg_name);
+    }
+    if (HL_TABLE()[idx].sg_rgb_sp_name != NULL) {
+      HL_TABLE()[idx].sg_rgb_sp = name_to_color(HL_TABLE()[idx].sg_rgb_sp_name);
```

Comment:
`HL_TABLE()[idx]` can be saved to a local variable to make this code a bit more readable.